File: jun/naq/stamp_brute.cpp
#include <bits/stdc++.h>                                                        
                                                                                
using namespace std;                                                            
#include<set>
                                                                                
void solve(vector<int> clusters, vector<int> queries) {                         
  int addLeft [clusters.size()] = {0};
  int addRight [clusters.size()] = {0};
  addLeft[0] = clusters[0];
  addRight[clusters.size() - 1] = clusters[clusters.size() - 1];
  for(int i = 1; i < clusters.size(); i++){
    addLeft[i] = addLeft[i-1] + clusters[i];
  }
  for(int i = clusters.size() - 2; i >= 0; i--){
    addRight[i] = addRight[i+1] + clusters[i];
  }

  set<int> s; 
  for(int i = 0; i < clusters.size(); i++){
    s.insert(addLeft[i]);
    s.insert(addRight[i]);
  }
  
  for(int i = 0; i < clusters.size(); i++) {
    for(int j = clusters.size() - 1; j > i; j--) {
      if(i + j <= clusters.size()){
        s.insert(addLeft[i] + addRight[j]); 
      }
    }
  }

  for(int i = 0; i < queries.size(); i++){
    if(s.find(queries[i]) != s.end() || queries[i] == 0) {
      cout << "Yes" << endl;
    } else {
      cout << "No" << endl;
    }
  }
}                                                                               

                                                                                
int main() {                                                                    
                                                                                
    int C = 1;                                                                  
    int Q = 1;                                                                  
    int q = 1;                                                                  
    int cluster;                                                                
    vector<int> clusters;                                                       
    cin >> C >> Q;                                                              
    for(int i = 0; i < C; i++) {                                                
      cin >> cluster;                                                           
      clusters.push_back(cluster);                                              
    }                                                                           
    vector<int> queries;                                                        
    while(Q--) {                                                                
      cin >> q;                                                                 
      queries.push_back(q);                                                     
    }                                                                           
    solve(clusters, queries);                                                   
    return 0;                                                                   
}        

File: jun/naq/stamp_brute.py
def solve(clusters, queries):
  query = max(queries)
  addLeft = []
  addRight = []
  addLeft.append(clusters[0])
  for i in range(1, len(clusters)):
    addLeft.append(clusters[i] + addLeft[i-1])
  rev = clusters[::-1]
  addRight.append(clusters[-1])
  for i in range(1, len(rev)):
    addRight.append(rev[i] + addRight[i-1])
  addRight = addRight[::-1]


  s = set() 
  for i in range(len(clusters)):
    s.add(addLeft[i])
    s.add(addRight[i])

  for i in range(len(clusters)):
    for j in range(len(clusters)-1, i, -1):
      s.add(addLeft[i] + addRight[j])
      if(addLeft[i] + addRight[i] > query):
        j = i 
    if addLeft[i] > query: 
      i = len(clusters)


  for i in queries:
    if i in s:
      print("Yes")
    elif i == 0:
      print("Yes")
    else:
      print("No")



C = input()
Q = int(C.split()[1])
C = int(C.split()[0])
clusters = input().split() 
for i in range(len(clusters)):
  clusters[i] = int(clusters[i])
queries = []
for i in range(Q):
  q = input() 
  queries.append(int(q))

solve(clusters, queries)
